docs/sources/alerting/fundamentals/notifications/message-templating.md

```diff
@@ -16,57 +16,43 @@ labels:
+- Add notification templates to contact points for reuse and consistent messaging in your notifications.
+- Use notification templates to change the title, message, and format of the message in your notifications.
 
-The following example shows the use of a custom template within one of the contact point fields.
+There are two ways to template your alert notification messages:  labels and annotations and notification templates.
 
-{{< figure src="/static/img/docs/alerting/unified/contact-points-use-template-8-0.png" class="docs-image--no-shadow" max-width= "550px" caption="Default template" >}}
+The below diagram illustrates the entire process of templating, from the creation of labels and annotations in alert rules or notification templates in contact points, and how they are when exported and applied to your alert notification messages.
 
-### Nested templates
+{{< figure src="/media/docs/alerting/{{templating-diagram.png" max-width="1200px" max-height="900px" caption="How Templating works" >}}
 
-You can embed templates within other templates.
+In this diagram:
 
-For example, you can define a template fragment using the `define` keyword:
+- **Monitored Application**: This could be a web server, a database, or any other service generating metrics. For example, it might be an NGINX server providing metrics about request rates, response times, etc.
+- **Prometheus**: Prometheus collects metrics from the monitored application. For example, it might scrape metrics from the NGINX server, including labels like instance (the server hostname) and job (the service name). See more details on the architecture. 
+- **Grafana**: Grafana queries Prometheus to retrieve metrics data. For example, you might create an alert rule to monitor NGINX request rates over time, and template labels or annotations based on the instance label.
+- **Alertmanager**: Part of the Prometheus ecosystem, Alertmanager handles alert notifications. For example, if the request rate exceeds a certain threshold on a particular NGINX server, Alertmanager can send an alert notification to, for example, Slack or email, including the server name and the exceeded threshold (the instance label will be interpolated, and the actual server name will appear in the alert notification).
+- **Alert notification**: When an alert condition is met, Alertmanager sends a notification to various channels such as Slack, Grafana OnCall, etc. These notifications can include information from the labels associated with the alerting rule. For example, if an alert triggers due to high CPU usage on a specific server, the notification message might include details like server name (instance label), disk usage percentage, and the threshold that was exceeded.
 
-```
-{{ define "mytemplate" }}
-  {{ len .Alerts.Firing }} firing. {{ len .Alerts.Resolved }} resolved.
-{{ end }}
-```
+## Next steps
```

Comment:
i don't think we need next steps here - because the two topics are nested anyway